app/views/observations/show/_observation_details.html.erb

```diff
@@ -17,22 +17,16 @@ caption ||= false
 
 <p class="obs-lat-lng" id="observation_lat_lng">
   <%=
-    hidden = "<i>(#{:show_observation_gps_hidden.t})</i>".html_safe
-    if obs.lat &&
-        (!obs.gps_hidden || obs.can_edit?)
+    if obs.reveal_location?
       link_to("#{obs.display_lat_long.t} " \
               "#{obs.display_alt.t} " \
               "[#{:click_for_map.t}]".html_safe,
               map_observation_path(id: obs.id))
-    elsif obs.lat
-      hidden + " ".html_safe + obs.display_alt.t
-    else
-      obs.display_alt.t
     end
-  %>
+    %>
   <%=
-    if obs.lat && obs.gps_hidden && obs.can_edit?
-      hidden
+    if obs.lat && obs.gps_hidden
+      "<i>(#{:show_observation_gps_hidden.t})</i>".html_safe
```

Comment:
@mo-nathan  - I just realized this partial, `observation_details`, is also called as a sub-partial on the obs index, because i added it to the lightbox caption.

So i'm refactoring the parts of the partial used in the lightbox caption as helpers.

Can you check that my refactor here is the same logically? One difference is i've got it returning nothing, instead of a blank `<p>`, if there's no latitude prop on the observation. 

```ruby

  def observation_details_where_gps(obs:)
    return "" unless obs.lat

    gps_display_link = link_to([obs.display_lat_long.t,
                                obs.display_alt.t,
                                "[#{:click_for_map.t}]"].safe_join(" "),
                               map_observation_path(id: obs.id))
    gps_hidden_msg = tag.i("(#{:show_observation_gps_hidden.t})")

    tag.p(class: "obs-where-gps", id: "observation_where_gps") do
      concat(gps_display_link) if !obs.gps_hidden || obs.can_edit?
      concat(gps_hidden_msg) if obs.gps_hidden
    end
  end
```

Please let me know any modifications if needed, and i'll fix my PR. Then you can revert this change in your PR to avoid a merge conflict.

Reply:
@mo-nathan — Aha. For your merge you'll have to change this helper to use your new Observation method `obs.reveal_location?`. I don't think i can add that method here because it calls your other new method `obs.project_admin?`.

Reply:
Thanks for mentioning the conflict.  Resolved.